### ISSUE TITLE
addGeneIntegrationMatrix error "Error occured and could not be resolved after 2 additional attempts"

### ISSUE THREAD

**Issue by hoholee** [GreenleafLab/ArchR#625]
Hi,

I tried to run the scATAC and scRNA integration following the tutorial and it went OK. But when I started to use my own data I ran into this error:

```
2021-03-04 09:34:40 : Block (1 of 12) : Seurat FindTransferAnchors, 4.729 mins elapsed.

2021-03-04 09:34:42 : FN : Error occured, attempting again (1 of 2)

2021-03-04 09:34:43 : FN : Error occured, attempting again (2 of 2)

2021-03-04 09:34:44 : FN : Error occured and could not be resolved after 2 additional attempts!
```

This error message does not look informative to me. Here is the full log file:
[ArchR-addGeneIntegrationMatrix-7d5b6d936832-Date-2021-03-04_Time-09-28-51.log](https://github.com/GreenleafLab/ArchR/files/6133647/ArchR-addGeneIntegrationMatrix-7d5b6d936832-Date-2021-03-04_Time-09-28-51.log)

Any idea what's going on?
Thanks in advance~

Best,
Junhao.

**Comment by rcorces** [comment]
I'm not sure what is causing this error. Perhaps @jgranja24 can weigh in. It would be helpful if you provided more details on what was done ___before___ you ran this command and how you created your `seRNA` object.

**Comment by hoholee** [comment]
I figured it out. My `seRNA` object is imported directly from a Seurat object, which was SCtransformed.
At first I was thinking maybe it's because I'm trying to integrate two huge datasets and the server ran out of memory. So I tried subsetting both the `archR` object and the `seRNA` object before integration. In this case R reported a more meaningful error message:
```
An SCT assay (SCT) was provided for reference.assay but normalization.method was set as LogNormalize.
```
But again this error message was not shown in the ArchR log file but instead similar errors as I posted earlier were reported:
```
2021-03-15 11:08:34 : Block (1 of 1) : Seurat FindTransferAnchors, 0.371 mins elapsed.

2021-03-15 11:08:35 : FN : Error occured, attempting again (1 of 2)

2021-03-15 11:08:36 : FN : Error occured, attempting again (2 of 2)

2021-03-15 11:08:37 : FN : Error occured and could not be resolved after 2 additional attempts!
```

I fixed the error by simply setting the default assay of the `seRNA` object to `RNA` so the log-normalized matrix in the `data` slot of the `RNA` assay is used in the integration:
```
DefaultAssay(seRNA) <- "RNA"
```

I guess a more elegant way is to use the `transferParams` parameter when running `addGeneIntegrationMatrix`?

Anyway, I would recommend improving the error report for `ArchR` so that it will help users to debug easier. Thanks!
